### PR TITLE
Add Varnish Cache in additional services docs section

### PR DESCRIPTION
**Pull request by aramalipoor**

### CODE REVIEW

File: additional-services/varnish.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Varnish Cache
+[Varnish Cache](https://varnish-cache.org) is a web application accelerator and is really, really fast. It typically speeds up delivery with a factor of 300 - 1000x, depending on your architecture.
+
+You can quickly deploy Varnish on AbarCloud using the following steps:
+1. From the dashboard, go to Add to Project > Browse Catalog > Utility > Varnish Cache.
```
Comment (aramalipoor) [inline additional-services/varnish.md]: @alikhajeh1 I think we need a new category?! It can be Cache (redis, varnish), or Utility, etc.

**Comment by alikhajeh1** [comment]
test comment

**Comment by alikhajeh1** [comment]
Test 2

**Comment by alikhajeh1** [comment]
@aramalipoor test

**Comment by alikhajeh1** [comment]
test 3

**Comment by alikhajeh1** [comment]
This comment

**Comment by alikhajeh1** [comment]
Test 5

**Comment by alikhajeh1** [comment]
test 7

**Comment by alikhajeh1** [comment]
test 8

**Comment by alikhajeh1** [comment]
test 9

**Comment by alikhajeh1** [comment]
test 10

File: additional-services/varnish.md (hunk 1 above)
Comment (aliscott) [reply additional-services/varnish.md]: We can create a cache category, not sure where redis fits though... it's a datastore that can be used as a cache

Comment (aramalipoor) [reply additional-services/varnish.md]: +1 for Cache categoy. I assume these categories work like tags, so it's going to show Redis in both Cache and Data Stores!?

File: additional-services/varnish.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Varnish Cache
+[Varnish Cache](https://varnish-cache.org) is a web application accelerator and is really, really fast. It typically speeds up delivery with a factor of 300 - 1000x, depending on your architecture.
+
+You can quickly deploy Varnish on AbarCloud using the following steps:
+1. From the dashboard, go to Add to Project > Browse Catalog > Cache > Varnish
+2. Fill in the form, and click on Create.
+3. Click on Continue to overview and wait for the pod to start running.
+4. Change your application Route (or [create a new route](https://docs.abarcloud.com/management/custom-domain-names.html#setting-up-a-custom-domain-for-your-service)) to point to the new Varnish Cache service.
```
Comment (aliscott) [inline additional-services/varnish.md]: Should we tell them how to change the route?

File: additional-services/varnish.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Customization
+Varnish Cache image in AbarCloud comes with pre-defined configurations which works for many applications.
```
Comment (aliscott) [inline additional-services/varnish.md]: The AbarCloud Varnish image includes sensible pre-defined configurations for many application use cases.

File: additional-services/varnish.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+For advanced usages we highly-recommend to fork [this repository] and change `default.vcl` based on your application needs.
+
+You can find many [useful examples](https://github.com/mattiasgeniar/varnish-4.0-configuration-templates/) of Varnish Cache configuration in web.
```
Comment (aliscott) [inline additional-services/varnish.md]: We are using 4.1 but this link is for 4.0. Will this work ok?

File: additional-services/varnish.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+You can quickly deploy Varnish on AbarCloud using the following steps:
+1. From the dashboard, go to Add to Project > Browse Catalog > Cache > Varnish
+2. Fill in the form, and click on Create.
+3. Click on Continue to overview and wait for the pod to start running.
+4. Change your application Route (or [create a new route](https://docs.abarcloud.com/management/custom-domain-names.html#setting-up-a-custom-domain-for-your-service)) to point to the new Varnish Cache service.
+
+## Customization
+Varnish Cache image in AbarCloud comes with pre-defined configurations which works for many applications.
+
+For advanced usages we highly-recommend to fork [this repository] and change `default.vcl` based on your application needs.
```
Comment (aliscott) [inline additional-services/varnish.md]: There's no link here for `[this repository]`

**Comment by alikhajeh1** [comment]
Lets continue to use relative links for everything inside docs so we can run it all locally to test easily

File: additional-services/varnish.md (hunk 4 above)
Comment (aramalipoor) [reply additional-services/varnish.md]: Actually `4.0` is not Varnish version. It's VCL version. So yeah it works for 4.x ;)

File: additional-services/varnish.md (hunk 5 above)
Comment (aramalipoor) [reply additional-services/varnish.md]: Oops good catch!

File: additional-services/varnish.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Varnish Cache
+[Varnish Cache](https://varnish-cache.org) is a web application accelerator and is really, really fast. It typically speeds up delivery with a factor of 300 - 1000x, depending on your architecture.
```
Comment (alikhajeh1) [inline additional-services/varnish.md]: Maybe add a note on when Varnish is suitable, and briefly explain what happens to dynamic parts (e.g. data fetched via APIs)

File: additional-services/varnish.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+You can quickly deploy Varnish on AbarCloud using the following steps:
+1. From the dashboard, go to Add to Project > Browse Catalog > Cache > Varnish
```
Comment (alikhajeh1) [inline additional-services/varnish.md]: See Slack note about categories

**Pull request merged**